stan/math/gpu/cholesky_decompose.hpp

```diff
@@ -16,16 +16,12 @@
 
 namespace stan {
 namespace math {
-inline matrix_gpu cholesky_decompose(matrix_gpu& A, const int block = 100,
-                                     const int divisor = 2,
-                                     const int min_block = 100);
+inline matrix_gpu cholesky_decompose(matrix_gpu& A, const int min_block);
 namespace internal {
 inline matrix_gpu cholesky_decompose_recursion(matrix_gpu& A,
-                                               const int block = 100,
-                                               const int divisor = 2,
-                                               const int min_block = 100) {
+                                               const int min_block) {
   matrix_gpu L(A.rows(), A.cols());
-  if (block <= min_block || divisor <= 1) {
+  if (A.rows() <= min_block || A.rows() < 100) {
```

Comment:
Why are we shortcircuiting `min_block` with a cap at 100 here?

Reply:
With really small `min_block` the execution becomes really slow, 100 is just an arbitrary value that gives good overall performance. 

Since `min_block` is defined as `A.rows * 0.03` this means that `min_block˙ is actually 100 for `A.rows() < 3333˙ and `A.rows()*0.03` for larger sizes. 

Would we prefer to have something more in the likes of

```
if(A.rows() > 1800 && A.rows() <= 2500)
{
min_block = 100;
} else if(A.rows() > 2500 && A.rows() < 3000)
{
min_block = 120;
}
...
```
I am fine either way.